### DISCUSSION TITLE
If the next-auth session cookie is httpOnly, how does next-auth know when it's logged in?

### DISCUSSION THREAD

**Discussion by yrokhlin** [nextauthjs/next-auth#3498] [Help]
Since Javascript can't touch and httpOnly cookie, how does next-auth know on the client that there is an active session?

**Comment by balazsorban44** [comment]
https://next-auth.js.org/getting-started/rest-api#get-apiauthsession
https://next-auth.js.org/getting-started/client#usesession
https://next-auth.js.org/getting-started/client#getsession

**Reply by yrokhlin** [reply]
No I understand which part of the next-auth API does this, I am asking specifically HOW that API works.
Like how does the actual mechanism work for detecting if the httpOnly cookie exists?
My apologies if the question was not immediately clear.

**Reply by balazsorban44** [reply]
look at the source code, it's open 😊
